### PR TITLE
Type checking: flow integration [1]

### PR DESCRIPTION
**Pull request by artkravchenko**

### CODE REVIEW

**Comment by indeyets** [comment]
@Oopscurity eslint-check fails. can you make it green? anyway, this patch is HUGE. would be much better to do it 1 step at a time. Can you reorder this branch and create a smaller pull-request, which:

1. adds tooling
2. enables flow-checks for some files and fixes automatic errors which show up after applying default rules (if there are such)

After that I would be glad to accept everything else as separate theme-based pull-requests

TL/DR: large pull-requests are bad

**Review by indeyets — CHANGES_REQUESTED** [review]
please separate into smaller pieces

**Comment by Oopscurity** [comment]
@indeyets There is a temporary state of the branch here to check out what most likely be in near future. I haven't sent changes related to https://github.com/Lokiedu/libertysoil-site/issues/670 before so I think it's important.

It's in progress, that's specified at the title. Of course, this branch should be separated into:

1. `flow-bin` and `eslint-flowtype` integration plus one or two examples
2. `babel-plugin-tcomb` integration (`tcomb`), necessary type definitions and few examples of run-time checking.
3. Gradual implementation of pages typecheck coverage.
4. Type definitions for common data types (that are in `src/prop-types` now) and gradual coverage of triggers and reducers.
5. Both static and dynamic typechecking the props of pages and components.
